Assets/Package/Runtime/Tooltip/TooltipHandler.cs

```diff
@@ -88,28 +56,38 @@ public override void Dispose()
             return factory.Create<T>(tooltipPrefab);
         }
 
-        public void ShowInstance(TooltipBehaviour instance, in TooltipData data)
+        public TooltipBehaviour ShowDefault(TooltipData data)
         {
-            var position = GetTooltipPosition(data.positioningType);
-            instance.UpdatePositionAndData(position, in data);
-            ShowInstance(instance);
+            return ShowTooltip<TooltipBehaviour>(data, null);
         }
 
-        public void ShowInstance(TooltipBehaviour instance)
+        public T ShowTooltip<T>(TooltipData data, T tooltipPrefab) where T : TooltipBehaviour
         {
-            ShowTooltip(instance);
+            var instance = GetInstance(tooltipPrefab);
+            ShowTooltip(instance, data);
+            return instance;
         }
 
-        public TooltipBehaviour ShowDefault(in TooltipData data)
+        public void ShowTooltip(TooltipBehaviour instance)
         {
-            return ShowTooltip<TooltipBehaviour>(in data, null);
+            ShowTooltip(instance, null);
         }
 
-        public T ShowTooltip<T>(in TooltipData data, T tooltipPrefab) where T : TooltipBehaviour
+        public void ShowTooltip(TooltipBehaviour instance, TooltipData data)
         {
-            var instance = GetInstance(tooltipPrefab);
-            ShowInstance(instance, in data);
-            return instance;
+            if (instance == null)
+            {
+                LogHandler.Log("[UI][Tooltip] Cannot show tooltip. Instance is null.", LogType.Error);
+                return;
+            }
+
+            HideTooltip();
+
+            var position = PointerPosition;
+            ActiveTooltip = instance;
+            ActiveTooltip.Prepare(data);
+            ActiveTooltip.UpdatePosition(position);
+            ActiveTooltip.Show();
         }
 
         public void HideTooltip()
```

Comment:
Wydaje mi się, że powinniśmy zwracać go do pooli dopiero po zakończeniu animacji ukrywania (może trwać jakiś czas). Wpiąłbym go więc do eventu zamiast wywoływać to bezpośrednio tutaj. W przypadku szybkiego przełączania tooltipów może nam nie zdążyć ukryć jednego i od razu go pokazać w innym miejscu bo już będzie w pooli.

Reply:
Chwilowo tooltipy nie są animowane i nie mogą być bo nie mają IActivityHandlera ale w sumie spoko, mogę to obsłużyć.